common/numerated/src/tree.rs

```diff
@@ -255,21 +253,24 @@ impl<T: Numerated> IntervalsTree<T> {
         }
     }
 
-    /// The same as [`Self::insert`], but returns [`I::Error`] if `try_into` [Interval] fails.
-    pub fn try_insert<I: TryInto<Interval<T>>>(&mut self, interval: I) -> Result<(), I::Error> {
-        let interval: Interval<T> = interval.try_into()?;
+    /// The same as [`Self::insert`], but returns [`I::Error`] if `try_into` [IntervalIterator] fails.
+    pub fn try_insert<I: TryInto<IntervalIterator<T>>>(
+        &mut self,
+        interval: I,
+    ) -> Result<(), I::Error> {
+        let interval: IntervalIterator<T> = interval.try_into()?;
         self.insert(interval);
         Ok(())
     }
 
     /// Remove `interval` from tree.
     /// - if `interval` is empty, then nothing will be removed.
-    /// - if `interval` is not empty, then after removing for each `p` ∈ `interval` ⇒ `p` ∉ `self`.
+    /// - if `interval` is not empty, then after removing: for each `p` ∈ `interval` ⇒ `p` ∉ `self`.
     ///
     /// Complexity: `O(log(n) + m)`, where
     /// - `n` is amount of intervals in `self`
     /// - `m` is amount of intervals in `self` ⋂ `interval`
-    pub fn remove<I: Into<Interval<T>>>(&mut self, interval: I) {
+    pub fn remove<I: Into<IntervalIterator<T>>>(&mut self, interval: I) {
```

Comment:
consider returning bool or rename to `clear() -> ()` mb

Reply:
What should we return than?
1) at least one point from `self` has been removed
2) each point from `interval` has been removed from `self`